src/node-jspm-jasmine.js

```diff
@@ -141,17 +179,79 @@ export function runTests(opts, errCallback = function() {}) {
 							importTheseTestFiles();
 						}
 					})
-					.catch(errCallback);
+					.catch((ex) => {
+						if (opts.coverage && opts.coverage.tempDirectory) {
+							// remove temporary directory
+							rimraf.sync(opts.coverage.tempDirectory)
+						}
+						errCallback(ex)
+					});
 				});
 			});
 		}
 	} catch(ex) {
-		removeTempFiles();
+		if (opts.coverage && opts.coverage.tempDirectory) {
+			// remove temporary directory
+			rimraf.sync(opts.coverage.tempDirectory)
+		}
 		errCallback(new Error(`Jasmine or Jspm may not be properly configured -- '${ex.toString()}'`));
 	}
 }
 
 function importTestFiles(SystemJS, jasmine, specDir, specFiles, coverage, errCallback) {
+
+	if (coverage) {
+		// jasmine will exit the process on its own, and since there doesn't appear
+		// to be a "onExit" hook, we're forced to intercept node's exit function and
+		// use as a "all done" jasmine hook.
+		// jasmine does offer a "addReporter" option, which contains a "jasmineDone"
+		// hook, however, it has no mechanism for waiting on async operations and
+		// async is needed for importing modules.
+		// Also, we should provide a "onExit" hook for the "programmatic" version,
+		// otherwise jasmine will abruptly kill the current running script.
+		const processExit = process.exit;
+		process.exit = function () {
+			// avoid misalignment with jasmine's output
+			console.log('');
+			// import the rest of the modules not already imported (evaluated)
+			// as dependencies of specs
+			Promise.all(Object.keys(coverage.coverageFiles).map(function (file) {
```

Comment:
@joeldenning I can't find it at the moment, but I read some open issue (maybe on Karma) about how the user said that coverage should analyze all the files it was told to process, and his reasoning was that, otherwise, it leaves him with a false sense of security about how tested the app is as a whole. I fully agree with his reasoning, and I would go as far as saying that coverage is not fully doing its job if it's not telling me about files that were not included in my specs. So I'm all for having it on by default. I think it's in the best interest of the user.